src/replication.c

```diff
@@ -757,14 +757,25 @@ create_replica_entry(spec_t *spec, int epfd, int mgmt_fd)
 	if (rc != 0) {
 		REPLICA_ERRLOG("pthread_mutex_init() failed err(%d) for "
 		    "replica(%s:%d)\n", rc, replica->ip, replica->port);
-		return NULL;
+		goto error;
```

Comment:
pthread_mutex_init failed, so, should not be doing pthread_mutex_destroy at 'error' label